typed/markers.go

```
@@ -0,0 +1,321 @@
+
+	// Allocate only as many walkers as needed for the depth by storing them here.
+	spareWalkers *[]*markerExtractorWalker
+	allocator    value.Allocator
+}
+
+func (v *markerExtractorWalker) prepareDescent(pe fieldpath.PathElement, tr schema.TypeRef) *markerExtractorWalker {
+	if v.spareWalkers == nil {
+		// first descent.
+		v.spareWalkers = &[]*markerExtractorWalker{}
+	}
+	var v2 *markerExtractorWalker
+	if n := len(*v.spareWalkers); n > 0 {
+		v2, *v.spareWalkers = (*v.spareWalkers)[n-1], (*v.spareWalkers)[:n-1]
+	} else {
+		v2 = &markerExtractorWalker{}
+	}
+	*v2 = *v
+	v2.typeRef = tr
+	v2.path = append(v2.path, pe)
+
+	// Track the element relationships of parents
+	v2.parentElementRelationship = append(v2.parentElementRelationship, v.resolveElementRelationship())
+	return v2
+}
+
+func (v *markerExtractorWalker) finishDescent(v2 *markerExtractorWalker) {
+	// if the descent caused a realloc, ensure that we reuse the buffer
+	// for the next sibling.
+	v.path = v2.path[:len(v2.path)-1]
+	v.parentElementRelationship = v2.parentElementRelationship[:len(v2.parentElementRelationship)-1]
+	*v.spareWalkers = append(*v.spareWalkers, v2)
+}
+
+func (v *markerExtractorWalker) extractMarkers() ValidationErrors {
+	return resolveSchema(v.schema, v.typeRef, v.value, v)
+}
+
+func (v *markerExtractorWalker) resolveElementRelationship() *schema.ElementRelationship {
+	if v.typeRef.ElementRelationship != nil {
+		return v.typeRef.ElementRelationship
+	} else if resolvedType, ok := v.schema.Resolve(v.typeRef); ok {
+		if resolvedType.Map != nil {
+			return &resolvedType.Map.ElementRelationship
+		} else if resolvedType.List != nil {
+			return &resolvedType.List.ElementRelationship
+		}
+	}
+	return nil
+}
+
+func (v *markerExtractorWalker) nearestElementRelationship() *schema.ElementRelationship {
+	for i := len(v.parentElementRelationship) - 1; i >= 0; i-- {
+		if v.parentElementRelationship[i] != nil {
+			return v.parentElementRelationship[i]
+		}
+	}
```

We're only looking for the nearestElementRelationship, what if it is nested? e.g. an atomic List which contains maps(not atomic), 
```
              types:
              - name: TestType
                map:
                  fields:
                  - name: items
                    type:
                      list:
                        elementType:
                          map:
                            fields:
                            - name: key
                              type:
                                scalar: string
                            - name: value
                              type:
                                scalar: string
                        elementRelationship: atomic
```
and applyconfig is `{"items": [{"key": "k1", "value": {"k8s_io__value": "unset"}}]}`, should we throw an error for it? The parentElementRelationship would be `["", "atomic", ""]` when validating the marker location.

Or, it should be `if v.parentElementRelationship[i] != nil &&  *v.parentElementRelationship[i] != "" && *v.parentElementRelationship[i] != "separable {`  to looking for the nearest "atomic" or 'associative"?